dev/src/Command/AddComponentCommand.php

```diff
@@ -184,21 +210,215 @@ protected function execute(InputInterface $input, OutputInterface $output)
+            $output->writeln("\n\nCopying the library code from bazel-bin");
+            $output->writeln($this->owlbotCopyBazelBin($new->componentName, $googleApisDir));
+            $output->writeln("\n\nOwlbot post processing");
+            $output->writeln($this->owlbotPostProcessor());
+        } else if ($input->getOption('googleapis-gen-path')) {
+            $this->checkDockerAvailable();
+            $googleApisGenDir = realpath($input->getOption('googleapis-gen-path'));
+            $output->writeln("\n\nCopying the library code from googleapis-gen");
+            $output->writeln($this->owlbotCopyCode($new->componentName, $googleApisGenDir));
+            $output->writeln("\n\nOwlbot post processing");
+            $output->writeln($this->owlbotPostProcessor());
+        }
+
         $output->writeln('');
         $output->writeln('');
         $output->writeln('Success!');
 
         return 0;
     }
 
+    private function validateOptions(InputInterface $input, OutputInterface $output): void
+    {
+        $error = '<error>googleapis-gen-path and bazel-path options ' .
+            'are mutually exclusive. Will build library with bazel.</error>';
+        if ($input->getOption('bazel-path') && $input->getOption('googleapis-gen-path')) {
+            $output->writeln('');
+            $output->writeln($error);
+            $output->writeln('');
+        }
+    }
+
+    private function loadYamlConfigContent(NewComponent $new, string $protoDir): ?array
+    {
+        $yamlFilePath = sprintf('%s/%s%s%s.yaml',
+            $protoDir,
+            strtolower($new->shortName),
+            $new->version ? '_' : '',
+            $new->version ?? ''
+        );
+
+        try {
+            return Yaml::parse($this->loadProtoContent($yamlFilePath));
+        } catch (Exception $e) {
+            // Handle error gracefully.
+            return null;
+        }
+    }
+
     private function loadProtoContent(string $proto): string
     {
         if (file_exists($proto)) {
             return file_get_contents($proto);
         }
         $protoUrl = 'https://raw.githubusercontent.com/googleapis/googleapis/master/' . $proto;
-        $client = new Client();
-        $response = $client->get($protoUrl);
+        $response = $this->httpClient->get($protoUrl);
         return (string) $response->getBody();
     }
+
+    private function bazelQueryAndBuildLibrary(string $protoPath, string $googleApisDir): string
+    {
+        $command = ['bazel', '--version'];
+        $output = $this->runCommand($command);
+        // Extract the version number from the output
+        $match = preg_match('/bazel\s+(?P<version>\d+\.\d+\.\d+)/', $output, $matches);
+        if (!$match || $matches['version'] !== self::BAZEL_VERSION) {
+            throw new RuntimeException('Bazel 6.0.0 is not available');
+        }
+
+        $componentPath = str_replace(['(', ')'], '', $protoPath);
```

Comment:
nit: instead of this, we should do what we did for `loadYamlConfigContent`, and pass in `dirname($proto)` and use the variable name `$protoDir`

Reply:
Here, the component path in the rule must be like `google/cloud/config/v1`. The `dirname($proto)` will have the full path as value like `../googleapis/google/cloud/config/v1`. I did not find anything useful in `NewComponent` as well (except the one I used in the current code).

Reply:
Sorry, I mean't to say use `dirname($protoFile)`. We strip the relative path for the `$proto` [earlier in the command](https://github.com/googleapis/google-cloud-php/blob/e0baa3b5328414268cd2ee1035f38b2c2706d55a/dev/src/Command/AddComponentCommand.php#L80C9-L80C19).

So if you use `dirname($protoFile)`, you'll get the expected result without having to strip parentheses.

Reply:
Done